src/integration-test/java/com/commercetools/project/sync/ProductSyncWithNestedReferencesIT.java

```diff
@@ -117,6 +132,26 @@ static void setupSourceProjectData(@Nonnull final SphereClient sourceProjectClie
             .attributeConstraint(AttributeConstraint.NONE)
             .build();
 
+    final AttributeDefinitionDraft setOfStateAttributeDef =
+        AttributeDefinitionDraftBuilder.of(
+                SetAttributeType.of(ReferenceAttributeType.of(State.referenceTypeId())),
+                "states",
+                ofEnglish("states"),
+                false)
+            .searchable(false)
+            .attributeConstraint(AttributeConstraint.NONE)
+            .build();
+
+    final AttributeDefinitionDraft setOfCustomerAttributeDef =
+        AttributeDefinitionDraftBuilder.of(
+                SetAttributeType.of(ReferenceAttributeType.of(Customer.referenceTypeId())),
+                "customers",
+                ofEnglish("customers"),
+                false)
+            .searchable(false)
+            .attributeConstraint(AttributeConstraint.NONE)
+            .build();
+
```

Comment:
The test case now can test set of reference type attributes. Do you think it is possible to test reference type (not inside a set) attributes?

Reply:
As discussed over call, We are already handling both ReferenceType Attributes and SetOfReferenceType Attributes in the same test case. So I think It is good for now.  Because Utils method is performing replacement only if the type is Product Else It will return the Reference Type as it is.